### PR TITLE
Argo Workflows Prometheus dashboard KPI and layout updates

### PR DESCRIPTION
**Pull request by algchoo**
### Description
This PR updates the layout and includes new KPIs for the Argo Workflows Prometheus dashboard.

### Changes
* [updated readme, metadata, and prometheus metadata to reflect additional metrics](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/commit/b4c6a030357bcc432442ac1459ac2209e65fc9f0)
* [updated dashboard with new layout and kpi metrics](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/commit/faa9c1e0da8b2561da90b8fec24f3017192d6768)
* [updated screenshot](https://github.com/GoogleCloudPlatform/monitoring-dashboard-samples/pull/724/commits/dfeb3b1bc379744131f894b3ac5137a5a5de45a1)

### Updated screenshot
argo-workflows-prometheus

### Additional KPIs
* Workflows With Pods Not Running
* Kuberenetes Request Rates

### CODE REVIEW

File: dashboards/argo-workflows/argo-workflows-prometheus.json [hunk 1]
```diff
@@ -17,115 +17,125 @@
...
+        "width": 8,
+        "height": 8,
         "widget": {
+          "title": "Pending",
           "scorecard": {
             "blankView": {},
+            "thresholds": [],
             "timeSeriesQuery": {
-              "prometheusQuery": "argo_workflows_count{status=\"Pending\",${Cluster},${Location},${Namespace}}"
+              "prometheusQuery": "argo_workflows_count{status=\"Pending\",${Cluster},${Location},${Namespace}}",
```
Comment (johnbryan) [inline dashboards/argo-workflows/argo-workflows-prometheus.json:33]: can all of these scorecards be wrapped in `sum(...)`?

I think there could hypothetically be pending/running/etc stuff on multiple pods, and we'd want the total across pods.

File: dashboards/argo-workflows/argo-workflows-prometheus.json [hunk 2]
```diff
@@ -17,115 +17,125 @@
...
             "dataSets": [
               {
+                "breakdowns": [],
+                "dimensions": [],
+                "measures": [],
                 "plotType": "LINE",
                 "targetAxis": "Y1",
                 "timeSeriesQuery": {
-                  "prometheusQuery": "argo_workflows_operation_duration_seconds_sum{${Cluster},${Location},${Namespace}}"
+                  "prometheusQuery": "argo_workflows_operation_duration_seconds_bucket{${Cluster},${Location},${Namespace}}",
```
Comment (johnbryan) [inline dashboards/argo-workflows/argo-workflows-prometheus.json:120]: _bucket metrics in promQL generally need some kind of aggregation applied to display well. What if we call this chart "Median Operation Duration" and use:

`histogram_quantile(0.50,rate(argo_workflows_operation_duration_seconds_bucket{${Cluster},${Location},${Namespace}}[${__interval}]))`

With the query as-is, we can see in the updated screenshot that the legend has a bunch of numbers which is kind of confusing to read - it's showing a separate line for how many operations happen in each duration bucket, with buckets like <.1 seconds, .1-.25 seconds, and finally up to +Infinity seconds.

I've found it helpful to build promQL for these bucket metrics by first building the query in the basic query builder mode, then converting it automatically to promQL.
image

File: dashboards/argo-workflows/argo-workflows-prometheus.json [hunk 3]
```diff
@@ -188,23 +208,97 @@
...
             "dataSets": [
               {
+                "breakdowns": [],
+                "dimensions": [],
+                "measures": [],
                 "plotType": "LINE",
                 "targetAxis": "Y1",
                 "timeSeriesQuery": {
-                  "prometheusQuery": "rate(argo_workflows_queue_latency_sum{${Cluster},${Location},${Namespace}}[1m])"
+                  "prometheusQuery": "rate(argo_workflows_queue_latency_bucket{${Cluster},${Location},${Namespace}}[1m])",
```
Comment (johnbryan) [inline dashboards/argo-workflows/argo-workflows-prometheus.json:217]: suggestion: Median Queue Latency

`histogram_quantile(0.50, sum by (queue_name, le)(rate(argo_workflows_queue_latency_bucket{${Cluster},${Location},${Namespace}}[${__interval}])))`

**Review by johnbryan — CHANGES_REQUESTED** [review]

File: dashboards/argo-workflows/argo-workflows-prometheus.json (hunk 2 above)
Comment (algchoo) [reply dashboards/argo-workflows/argo-workflows-prometheus.json]: @johnbryan Thanks for the explanation here, it was helpful and I appreciate it. I've updated both the queries.

**Review by johnbryan — APPROVED** [review]
Looks great, thanks!

**Pull request merged**